packages/react/src/components/header/components/withDropdown/WithDropdown.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { ElementType, HTMLAttributes, useEffect, useRef, useState } from 'react';
+
+import { IconCross } from '../../../../icons';
+import { IconButton } from '../headerActionBar/HeaderActionBarItem';
+import classNames from '../../../../utils/classNames';
+import classes from './WithDropdown.module.scss';
+
+export interface WithDropdownPropsType extends HTMLAttributes<HTMLDivElement> {
+  id: string;
+  label: string;
+  icon: ElementType;
+  children: JSX.Element;
+}
+
+export const WithDropdown = ({ id, children, label, icon, ...props }: WithDropdownPropsType): JSX.Element => {
+  const containerElementRef = useRef<HTMLDivElement>(null);
+  const [visible, setDisplayProperty] = useState(false);
+
+  const getContainer = () => containerElementRef.current as HTMLDivElement;
+
+  const handleButtonClick = () => {
+    setDisplayProperty(!visible);
+  };
+  const handleDocumentClick = (event) => {
+    const container = getContainer();
+    const eventTargetNode = event.target;
+    if (!container.contains(eventTargetNode)) {
+      setDisplayProperty(false);
+    }
+  };
+  const handleBlur = (event) => {
+    const container = getContainer();
+    const eventTargetNode = event.relatedTarget;
+    if (!container.contains(eventTargetNode)) {
+      setDisplayProperty(false);
+    }
+  };
```

Comment:
This actually works differently from hel.fi implementation. There the dropdowns stay open even when user tabs to a different navigation item. But I think this is actually better? Let's see what Unicus says also :)